tests/Microsoft.Bot.Builder.Tests/AppInsight_MiddlewareTests.cs

```diff
@@ -0,0 +1,293 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.Tests
+{
+    [TestClass]
+    public class AppInsight_MiddlewareTests
+    {
+        private List<ITelemetry> _sendItems;
+        private TelemetryConfiguration _configuration;
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _configuration = new TelemetryConfiguration();
+            _sendItems = new List<ITelemetry>();
+            _configuration.TelemetryChannel = new StubTelemetryChannel { OnSend = item => _sendItems.Add(item) };
+            _configuration.InstrumentationKey = Guid.NewGuid().ToString();
+            _configuration.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+            
+        }
+
+        [TestMethod]
+        [TestCategory("Middleware")]
+        public async Task AppInsight_LogActivities()
+        {
+            var transcriptStore = new MemoryTranscriptStore();
+            TestAdapter adapter = new TestAdapter()
+                .Use(new AppInsightsLoggerMiddleware(Guid.NewGuid().ToString(), false, false, _configuration));
+            //string instrumentationKey, bool logUserName = false, bool logOriginalMessage = false)
+            string conversationId = null;
+
+            await new TestFlow(adapter, async (context) =>
+            {
+                conversationId = context.Activity.Conversation.Id;
+                var typingActivity = new Activity
+                {
+                    Type = ActivityTypes.Typing,
+                    RelatesTo = context.Activity.RelatesTo
+                };
+                await context.SendActivityAsync(typingActivity);
+                await Task.Delay(500);
```

Comment:
You probably copy/pasted this `Task.Delay` from another test, but if you don't need it we should remove it. It slows down unit test execution for no reason. These tests alone (as they stand right now) will add 3s to running the tests when they should only cost us a couple ms at most.

/cc @cleemullins - we should revisit all tests that do this, it's a bad idea.